File: 2216-delete-the-middle-node-of-a-linked-list/2216-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        ListNode *mid;
        ListNode *ptr;
        ptr=head;
        mid=head;
        int n=0;
        int midIndex=0;
        ptr=ptr->next;
        if(ptr==NULL)
            return NULL;
        while(ptr->next!=NULL)
        {
            ptr=ptr->next;
            n++;
            if(n%2==0)
            {
                midIndex++;
                mid=mid->next;
            }
        }
        
            mid->next=mid->next->next;
        return head;
    }
};

File: 2216-delete-the-middle-node-of-a-linked-list/2216-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        if head.next==None:
            return None
        ptr = head.next
        mid = head
        n=0
        while ptr.next:
            ptr=ptr.next
            n+=1
            if n%2 == 0:
                mid=mid.next
        mid.next=mid.next.next
        return head
        